### ISSUE TITLE
NLP Sentiment Analysis

### ISSUE THREAD

**Issue by AHARRISRAGAVENDRA** [AHARRISRAGAVENDRA/HARRIS#1]
import nltk
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score

# Download required resources
nltk.download('movie_reviews')
from nltk.corpus import movie_reviews

# Load the dataset
documents = [(list(movie_reviews.words(fileid)), category)
             for category in movie_reviews.categories()
             for fileid in movie_reviews.fileids(category)]

# Convert list of words into sentences
data = [" ".join(words) for words, label in documents]
labels = [label for words, label in documents]

# Split data into training and testing
X_train, X_test, y_train, y_test = train_test_split(data, labels, test_size=0.2, random_state=42)

# Convert text into feature vectors
vectorizer = CountVectorizer()
X_train_counts = vectorizer.fit_transform(X_train)
X_test_counts = vectorizer.transform(X_test)

# Train the classifier
model = MultinomialNB()
model.fit(X_train_counts, y_train)

# Make predictions
y_pred = model.predict(X_test_counts)

# Evaluate the model
accuracy = accuracy_score(y_test, y_pred)
print("Model Accuracy:", round(accuracy * 100, 2), "%")

# Test with a custom sentence
sample_text = ["This movie was amazing and full of great moments!"]
sample_vec = vectorizer.transform(sample_text)
prediction = model.predict(sample_vec)
print("Sentiment:", prediction[0])
